client/client-lib/src/lib.rs

```diff
@@ -505,6 +505,21 @@ impl Client<UserClientConfig> {
 
         Ok(OutPoint { txid, out_idx: 0 })
     }
+
+    /// Notify gateway that we've escrowed tokens they can claim by routing our payment and wait
+    /// for them to do so
+    pub async fn await_outgoing_contract_execution(&self, contract_id: ContractId) -> Result<()> {
+        let gateway = self.fetch_gateway().await?;
+
+        let future = reqwest::Client::new()
+            .post(&format!("{}/pay_invoice", gateway.api))
+            .json(&contract_id)
+            .send();
+        minimint_api::task::timeout(Duration::from_secs(15), future)
```

Comment:
The timeout should be coordinated with the gateway imo. Many LN payments will take longer, the question is how long the GW will try. We should only time out if the gateway said it would try for time t and >t time passed.

Reply:
Made an [issue](https://github.com/fedimint/minimint/issues/301) for this. Will address in a separate PR.